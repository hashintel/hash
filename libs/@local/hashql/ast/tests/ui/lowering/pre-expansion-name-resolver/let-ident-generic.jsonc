//@ run: pass
//@ description: if the first argument cannot be interpreted as an ident, do nothing
["let", "a<T>", "b", ["+", "a", "c"]]
