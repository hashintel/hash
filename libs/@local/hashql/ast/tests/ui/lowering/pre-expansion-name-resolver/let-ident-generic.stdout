Expr#4294967040@30
  ExprKind (Call)
    CallExpr#4294967040@30
      Expr#4294967040@3
        ExprKind (Path)
          Path#4294967040@3 (rooted: true)
            PathSegment#4294967040@2 (name: kernel)
            PathSegment#4294967040@2 (name: special_form)
            PathSegment#4294967040@2 (name: let)
      Argument#4294967040@12
        Expr#4294967040@12
          ExprKind (Path)
            Path#4294967040@12 (rooted: false)
              PathSegment#4294967040@11 (name: a)
                PathSegmentArgument (GenericArgument)
                  GenericArgument#4294967040@10
                    Type#4294967040@9
                      TypeKind (Path)
                        Path#4294967040@9 (rooted: false)
                          PathSegment#4294967040@8 (name: T)
      Argument#4294967040@16
        Expr#4294967040@16
          ExprKind (Path)
            Path#4294967040@16 (rooted: false)
              PathSegment#4294967040@15 (name: b)
      Argument#4294967040@29
        Expr#4294967040@29
          ExprKind (Call)
            CallExpr#4294967040@29
              Expr#4294967040@20
                ExprKind (Path)
                  Path#4294967040@20 (rooted: true)
                    PathSegment#4294967040@19 (name: math)
                    PathSegment#4294967040@19 (name: add)
              Argument#4294967040@24
                Expr#4294967040@24
                  ExprKind (Path)
                    Path#4294967040@24 (rooted: false)
                      PathSegment#4294967040@23 (name: a)
              Argument#4294967040@28
                Expr#4294967040@28
                  ExprKind (Path)
                    Path#4294967040@28 (rooted: false)
                      PathSegment#4294967040@27 (name: c)
